kernel/src/main/scala/cats/kernel/Comparison.scala

```diff
@@ -0,0 +1,19 @@
+package cats.kernel
+
+/** ADT encoding the possible results of a comparison */
+sealed abstract class Comparison extends Product with Serializable {
+  /** The signum of this comparison */
+  def toInt: Int = this match {
+    case Comparison.GreaterThan => 1
+    case Comparison.EqualTo     => 0
+    case Comparison.LessThan    => -1
+  }
```

Comment:
It may be different with a `val` like that, but fwiw, at one point I benchmarked having scalaz's `Maybe` implement `fold` via pattern-matching in the parent vs dynamic dispatch to the child implementation and the former seemed to have a slight edge. It may be worth throwing together a benchmark if this is something that we care about. (I'm okay with it not being something that we care about, since people can always use the `Int` version of compare if they really want performance.)